app/src/main/java/com/example/poste/models/User.java

```diff
@@ -302,4 +304,62 @@ private String parseError(Response<ResponseBody> response) {
         }
         return null;
     }
+
+    /**
+     * Adds a new post the given folder
+     *
+     * @param folder folder to add the new post to
+     */
+    public void createNewPost(Folder folder, Context context, String title, String description, String url) {
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.createPost(getTokenHeader() ,new CreatePost(title, description, url, folder.getTitle()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                if (response.isSuccessful()) {
+                    updateFoldersAndPosts(context);
+                }
+                else {
+                    try {
+                        Toast.makeText(context, response.errorBody().string(), Toast.LENGTH_LONG);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
```

Comment:
Careful with these -- it'll crash the app. You could use a generic error message, or try to parse it out -- mind how the response is given from the backend. Or, consider using parseError -- a method also in User. It is setup to parse error messages from another API call. Assuming the responses are similar, it may work for this, too. The idea is they should all respond the same way, but I can't guarantee all implementations are the same (yet...)

At the very least, if trying to parse the error message fails, I'd suggest making a generic toast (and please log the response) rather than crashing

Reply:
Switched to generic error toast.